primitives/src/types.rs

```diff
@@ -67,3 +68,26 @@ pub type Header = generic::Header<BlockNumber, BlakeTwo256>;
 pub type Block = generic::Block<Header, OpaqueExtrinsic>;
 /// Block ID.
 pub type BlockId = generic::BlockId<Block>;
+
+/// The outer `FeeExchange` type. It is versioned to provide flexbility for future iterations
+/// while maintaining backward compatability.
+#[derive(PartialEq, Eq, Clone, Encode, Decode)]
+pub enum FeeExchange<T> {
+	/// A V1 FeeExchange, it may be `None` meaning no fee exchange is required
+	#[codec(compact)]
+	V1(FeeExchangeV1<T>),
+}
+
+/// A v1 FeeExchange
+/// Signals a fee payment requiring the CENNZX-Spot exchange. It is intended to
+/// embed within CENNZnet extrinsic payload.
+/// It specifies input asset ID and the max. limit of input asset to pay
+#[derive(PartialEq, Eq, Clone, Encode, Decode)]
+pub struct FeeExchangeV1<T> {
```

Comment:
I'm thinking we should drop the generic `T` here as it was really a poor attempt at versioning in it's previous incarnation. Since `V1` is its own version now it should not change (that would be a new version) so we can hard code the `Balance` type like `AssetId` is.
`pub max_payment: Balance`